### PR TITLE
add components to allow aggregating global counters and displaying a …

### PR DESCRIPTION
**Pull request by kroepke**
…per second rate of them

also enable to show more metrics in the browser (our plugins use org.graylog instead of org.graylog2)

### CODE REVIEW

**Comment by edmundoa** [comment]
Looks good, but there are some code style issues, will nitpick a bit :)

File: graylog2-web-interface/src/components/metrics/CounterRate.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import numeral from 'numeral';
+
+var CounterRate = React.createClass({
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/CounterRate.jsx]: Please use const to store the object: http://eslint.org/docs/rules/no-var

File: graylog2-web-interface/src/components/metrics/CounterRate.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  _placeholder() {
+    if (this.props.hideOnZero) {
+      return null;
+    }
+    return (<span>{this._prefix()}Calculating...</span>);
+  },
+
+  _prefix() {
+    if (this.props.prefix) {
+      return this.props.prefix + ' ';
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/CounterRate.jsx]: Please use templates for concatenating strings: http://eslint.org/docs/rules/prefer-template

File: graylog2-web-interface/src/components/metrics/CounterRate.jsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    }
+    return null;
+  },
+
+  _suffix() {
+    if (this.props.suffix) {
+      return ' ' + this.props.suffix;
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/CounterRate.jsx]: Same as above

File: graylog2-web-interface/src/components/metrics/CounterRate.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      if (!this._checkPrevMetric()) {
+        return this._placeholder();
+      }
+    }
+    const count = this.props.metric.count;
+
+    let rate = null;
+    if (this._checkPrevMetric()) {
+      const rateNum = (count - this.state.prevMetric.count) / (this.state.nowTs - this.state.prevTs);
+      rate = (<span key='rate' className="number-format">{this._prefix()}{numeral(rateNum).format('0,0')}{this._suffix()}</span>);
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/CounterRate.jsx]: Please use double quotes on JSX props.

File: graylog2-web-interface/src/components/metrics/CounterRate.jsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+    if (this.props.hideOnMissing && !rate) {
+      return null;
+    }
+
+    return (<span>
+      {rate}
+      {this.props.showTotal && <span key='absolute' className="number-format"> ({numeral(count).format('0')} total)</span>}
+      </span>);
+  }
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/CounterRate.jsx]: Missing comma: http://eslint.org/docs/rules/comma-dangle

File: graylog2-web-interface/src/components/metrics/MetricContainer.jsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import String from 'string';
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/MetricContainer.jsx]: Unused import

File: graylog2-web-interface/src/components/metrics/MetricContainer.jsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    zeroOnMissing: React.PropTypes.bool
+  },
+  mixins: [InjectActions('Metrics'), InjectStore('Metrics')],
+  componentWillMount() {
+    this.MetricsActions.addGlobal(this.props.name);
+  },
+  componentWillUnmount() {
+    this.MetricsActions.removeGlobal(this.props.name);
+  },
+  getDefaultProps() {
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/MetricContainer.jsx]: `componentWillMount` and `componentWillUnmount` should go after `getDefaultProps` and `getInitialState`.

File: graylog2-web-interface/src/components/metrics/MetricContainer.jsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import String from 'string';
+
+import { Spinner } from 'components/common';
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/MetricContainer.jsx]: Another unused import

File: graylog2-web-interface/src/components/metrics/MetricContainer.jsx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  render() {
+    if (!this.state.metrics) {
+      return (<span>Loading...</span>);
+    }
+    const fullName = this.props.name;
+    const throughput = Object.keys(this.state.metrics)
+      .map(nodeId => MetricsExtractor.getValuesForNode(this.state.metrics[nodeId], { throughput: fullName }))
+      .reduce((one, two) => one.throughput + two.throughput);
+    if (this.props.zeroOnMissing && (!throughput || !throughput.throughput)) {
+      throughput["throughput"] = 0;
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/MetricContainer.jsx]: Please use the dot notation for accessing known properties: http://eslint.org/docs/rules/dot-notation

File: graylog2-web-interface/src/components/metrics/MetricContainer.jsx [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+import String from 'string';
+
+import { Spinner } from 'components/common';
+
+import InjectStore from 'injection/InjectStore';
+import InjectActions from 'injection/InjectActions';
+
+import MetricsExtractor from 'logic/metrics/MetricsExtractor';
+
+var MetricContainer = React.createClass({
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/MetricContainer.jsx]: Please use const in here :)

File: graylog2-web-interface/src/components/metrics/MetricContainer.jsx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+      throughput["throughput"] = 0;
+    }
+    return (<div>
+      {
+        React.Children.map(this.props.children, function (child) {
+          return React.cloneElement(child, { metric: { full_name: fullName, count: throughput.throughput }})
+        })
+      }
+    </div>);
+  }
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/MetricContainer.jsx]: Another missing comma

File: graylog2-web-interface/src/components/metrics/MetricContainer.jsx [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+    const fullName = this.props.name;
+    const throughput = Object.keys(this.state.metrics)
+      .map(nodeId => MetricsExtractor.getValuesForNode(this.state.metrics[nodeId], { throughput: fullName }))
+      .reduce((one, two) => one.throughput + two.throughput);
+    if (this.props.zeroOnMissing && (!throughput || !throughput.throughput)) {
+      throughput["throughput"] = 0;
+    }
+    return (<div>
+      {
+        React.Children.map(this.props.children, function (child) {
```
Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/MetricContainer.jsx]: We should declare `this.props.children` in the propTypes

Comment (edmundoa) [inline graylog2-web-interface/src/components/metrics/MetricContainer.jsx]: Also please use arrow notation for that function

**Comment by bernd** [comment]
LGTM and tested. :+1:

**Pull request merged**